docs/Architecture.md

```diff
@@ -0,0 +1,243 @@
+# Architecture
+
+This document will cover the core Architectural decisions made for this UI, how aspects of the UI integrate together and work, as well as detail how the UI then integrates with a Strimzi deployment.
+
+## Contents
+
+- [Overview and objectives](#overview-and-objectives)
+- [Security](#security)
+- [Topology](#topology)
+  - [Production](#production-topology)
+  - [Development and test](#development-and-test-topology)
+- [Transport layer](#transport-layer)
+- [Strimzi integration](#strimzi-integration)
+- [Implementation details](#implementation-details)
+  - [Front end (client)](#client)
+    - [Technologies and patterns used](#technologies-and-patterns-used)
+    - [File name conventions](#file-name-conventions)
+    - [Routing and navigation model](#routing-and-navigation)
+    - [Component topology](#component-topology)
+  - [Back end (server)](#server)
+  - [Configuration and feature flagging](#configuration-and-feature-flagging)
+- [Code structure](#code-structure)
+  - [Directory structure](#directory-structure)
+  - [`Element`, `Group`, `Panel`, `Bootstrap` component pattern](#element-group-panel-bootstrap-component-pattern)
+  - [Swap-able view layers](#swap-able-view-layers)
+- [Supporting utilities and tools](#supporting-utilities-and-tools)
+  - [Storybook](#storybook)
+  - [Mock admin server](#mock-admin-server)
+
+### Overview and objectives
+
+The purpose of this UI is to provide a user of Strimzi a delightful Kafka administration user experience. To achieve this, focus of course needs to be placed on things like the design to make the UI intuitive to use. However, user experience is more than just the design: it also is the code, and it's ability to enable those experiences. The following documentation will detail the deliberate design, engineering and behavioural choices made to achieve this objective.
+
+In addition, we have set the following behaviours/goals/considerations for all development done in this repository:
+
+- Focusing on the user, be they developer or end user, at all times
+- To make the codebase as approachable as possible through the use of consistent and intuitive style, and supporting documentation
+- To make the codebase extensible and configurable
+- Adopting a cloud like CI/CD development model: lots of small and often changes, appropriately flagged, with the `master` branch being shippable at any time
```

Comment:
```suggestion
- Adopting a cloud like CI/CD development model: lots of small and often changes, appropriately flagged, with the `primary` branch being deployable at any time
```

Reply:
This will need to change pending the result of https://github.com/strimzi/strimzi-kafka-operator/issues/3289 (as I expect that outcome will effect all Strimzi repos). Until that time, I would leave it as `master` (again, matching existing Strimzi repos).

Reply:
Going back to this ... as far as I understood, GitHub will be changing later this year `master` to `main` ar least for new repos. So if you want to go with some new name from the beginning, it might make sense to use `main`. But up to you ... we can change it later when/if we change the branches in other projects.